### PR TITLE
CLDSRV-741: Fix & test AbortMPU edge cases orphan & race condition

### PR DESCRIPTION
**Pull request by BourgoisMickael**
Relates to [CLDSRV-669](https://scality.atlassian.net/browse/CLDSRV-669)

Uses https://github.com/scality/Arsenal/pull/2526

You can review by commit

[CLDSRV-669]: https://scality.atlassian.net/browse/CLDSRV-669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/api/apiUtils/object/abortMultipartUpload.js [hunk 1]
```diff
@@ -161,9 +163,13 @@ function abortMultipartUpload(authInfo, bucketName, objectKey, uploadId, log,
...
                 versionId: objectMD.versionId,
             });
-            return metadata.deleteObjectMD(bucketName, objectKey, {
-                versionId: objectMD.versionId,
-            }, log, err => {
+
+            const options = preprocessingVersioningDelete(
+                bucketName, destBucket, objectMD, objectMD.versionId, config.nullVersionCompatMode);
+            options.replayId = uploadId;
+            options.doesNotNeedOpogUpdate = true;
```
Comment (ghost) [inline lib/api/apiUtils/object/abortMultipartUpload.js:171]: Can we add a comment, just explaining it means in this cleanup case, this is a ghost (md, but no data, yet not a data loss), so we do not want to affect the backend services with useless oplog events, hence this option set to true?

**Review by ghost — APPROVED** [review]
Thanks!

File: lib/api/apiUtils/object/abortMultipartUpload.js (hunk 1 above)
Comment (BourgoisMickael) [reply lib/api/apiUtils/object/abortMultipartUpload.js:171]: [b41e77e](https://github.com/scality/cloudserver/pull/5948/commits/b41e77ec48bd1fe14965c544b8b9cb468e33c375)

File: tests/functional/aws-node-sdk/test/object/abortMPU.js [hunk 2]
```diff
@@ -315,35 +319,7 @@ describe('Abort MPU - Versioned Bucket Cleanup', function testSuite() {
...
-                    UploadId: upload.UploadId,
-                }).promise().catch(err => {
-                    if (err.code !== 'NoSuchUpload') {
-                        throw err;
-                    }
-                }),
-            ));
-
-            // Clean up all object versions
-            const listVersionsResponse = await s3.listObjectVersions({ Bucket: bucketName }).promise();
```
Comment (dvasilas) [inline tests/functional/aws-node-sdk/test/object/abortMPU.js:333]: We don't need this anymore?

**Review by dvasilas — APPROVED** [review]

File: tests/functional/aws-node-sdk/test/object/abortMPU.js (hunk 2 above)
Comment (BourgoisMickael) [reply tests/functional/aws-node-sdk/test/object/abortMPU.js:333]: It's been moved into the function `cleanupVersionedBucket` as the same pattern is reused in `after` hooks in multiple describes

Comment (dvasilas) [reply tests/functional/aws-node-sdk/test/object/abortMPU.js:333]: I thought that we didn't move `listObjectVersions` -> delete, but actually `empty` does that, thanks

**Review by nicolas2bert — APPROVED** [review]

**Comment by BourgoisMickael** [comment]
@bert-e force_reset
